x-pack/plugins/data_enhanced/server/search/session/monitoring_task.ts

```diff
@@ -48,12 +52,17 @@ function searchSessionRunner(
             logger,
           },
           sessionConfig
-        );
+        )
+          .pipe(takeUntil(aborted$))
```

Comment:
The cancelation idea is the following: 

We can use `pageConfig` to process sessions page by page. When abort is triggered we will process the last received page and won't fetch for the one that are left